redfish-core/lib/assembly.hpp

```diff
@@ -0,0 +1,494 @@
+    });
+}
+
+/**
+ * @brief Api to get assembly endpoints from mapper.
+ * @param[in] asyncResp - Shared pointer for asynchronous calls.
+ * @param[in] chassisPath - Chassis to which the assemblies are
+ * associated.
+ *
+ * @return None.
+ */
+inline void
+    getAssemblyEndpoints(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                         const std::string& chassisPath)
+{
+    BMCWEB_LOG_DEBUG("Get assembly endpoints");
+
+    sdbusplus::message::object_path assemblyPath(chassisPath);
+    assemblyPath /= "assembly";
+
+    // if there is assembly association, look
+    // for endpoints
+    dbus::utility::getAssociationEndPoints(
+        assemblyPath, [asyncResp, chassisPath](
+                          const boost::system::error_code& ec,
+                          const dbus::utility::MapperEndPoints& endpoints) {
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR("DBUS response error : {}", ec.value());
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        auto sortedAssemblyList = endpoints;
+        std::sort(sortedAssemblyList.begin(), sortedAssemblyList.end());
+
+        checkAssemblyInterface(asyncResp, chassisPath, sortedAssemblyList);
+        return;
+    });
+}
+
+/**
+ * @brief Api to check for assembly associations.
+ * @param[in] asyncResp - Shared pointer for asynchronous calls.
+ * @param[in] chassisPath - Chassis to which the assemblies are
+ * associated.
+ *
+ * @return None.
+ */
+inline void checkForAssemblyAssociations(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& chassisPath, const std::string& service)
+{
+    BMCWEB_LOG_DEBUG("Check for assembly association");
+
+    using associationList =
+        std::vector<std::tuple<std::string, std::string, std::string>>;
+
+    crow::connections::systemBus->async_method_call(
+        [asyncResp,
+         chassisPath](const boost::system::error_code ec,
+                      const std::variant<associationList>& associations) {
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR("DBUS response error : {}", ec.value());
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        const associationList* value =
+            std::get_if<associationList>(&associations);
+        if (value == nullptr)
+        {
+            BMCWEB_LOG_ERROR(
+                "DBUS response error. Received an empty association list.");
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        bool isAssemblyAssociation = false;
+        for (const auto& listOfAssociations : *value)
+        {
+            if (std::get<0>(listOfAssociations) != "assembly")
+            {
+                // implies this is not an assembly
+                // association
+                continue;
+            }
+
+            isAssemblyAssociation = true;
+            break;
+        }
+
+        if (isAssemblyAssociation)
```

Comment:
L338-L352 seems checking the given associationList   is an `assembly` or not.  
I think it can be simplified.

```
   #include <algorithm>
   #include <ranges>
    auto iter = std::ranges::find_if(value->begin(), value->end(), 
                          [](const auto& listOfAssociations) { 
                                  return std::get<0>(listOfAssociations) == "assembly";
                            });
     if(iter != value->end()
     {
              getAssemblyEndpoints(asyncResp, chassisPath);
     }
```